src/controller/BUILD.gn

```diff
@@ -24,41 +24,91 @@ source_set("nodatamodel") {
   public_deps = [ "${chip_root}/src/app" ]
 }
 
+CHIP_CONTROLLER_SOURCES = [
+  "AbstractDnssdDiscoveryController.cpp",
+  "AutoCommissioner.cpp",
+  "CHIPCommissionableNodeController.cpp",
+  "CHIPDeviceControllerFactory.cpp",
+  "CHIPDeviceControllerFactory.h",
+  "CommissioneeDeviceProxy.cpp",
+  "CommissionerDiscoveryController.cpp",
+  "CommissionerDiscoveryController.h",
+  "CommissioningDelegate.cpp",
+  "ExampleOperationalCredentialsIssuer.cpp",
+  "ExampleOperationalCredentialsIssuer.h",
+  "SetUpCodePairer.cpp",
+]
+
+CHIP_READ_CLIENT_SOURCES = [
+  "CHIPDeviceController.cpp",
+  "CommissioningWindowOpener.cpp",
+  "CommissioningWindowOpener.h",
+  "CurrentFabricRemover.cpp",
+  "CurrentFabricRemover.h",
+]
+
+if (!chip_controller || !chip_build_controller) {
+  # This source set exists specifically to deny including them without dependencies
+  # if "controller" sources do not contain them
+  #
+  # They are NOT intended to be used directly
+  #
+  # Also note: these source_sets are NOT compilable/usable.
+  #            expect tools like clang-tidy to fail to find include paths if these
+  #            are enabled.
+  source_set("gen_check_chip_controller_sources") {
+    sources = CHIP_CONTROLLER_SOURCES
+  }
+}
+
+if (!chip_controller || !chip_build_controller || !chip_enable_read_client) {
+  source_set("gen_check_chip_read_client_sources") {
+    sources = CHIP_READ_CLIENT_SOURCES
+  }
+}
+
+source_set("delegates") {
```

Comment:
this seems like an overly broad name, but I'm not totally sure how it gets namespaced with the build files - will this cause collisions if there is a combined server/controller device with server delegates?

Reply:
The namespacing is based on the build file, so the full name of `src/controller:delegates` will be used. It would not conflict with any other potential name of `delegates` since `src/controller:delegates` will be different than `foo/bar:delegates`.